docs/src/variables.md

```diff
@@ -256,9 +256,11 @@ julia> JuMP.lower_bound.(x)
 
 JuMP will form an `Array` of JuMP variables when it can determine at compile
 time that the indices are one-based integer ranges. Therefore `x[1:b]` will
-work, but `x[a:b]` will throw an error.
+work, but `x[a:b]` will not. If JuMP cannot determine that the indices are
```

Comment:
This wording is still ambiguous. "works" in this sentence means provides sufficient information for JuMP to conclude that it's correct to create an `Array`.

Maybe we need a table with different examples of what index sets will default to which containers (and which are also possible with the `container=` keyword). (Possible TODO.)